### PR TITLE
Updated for nightly

### PR DESCRIPTION
**Pull request by gsingh93**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -25,15 +66,15 @@ pub mod mock {
...
-        use std::io::Read;
         use std::net::SocketAddr;
 
         /// Create a new mock Request with the given method, url, and data.
-        pub fn new<'a, R>(method: method::Method, path: Url,
-                          data: &'a mut R) -> Request<'a>
-        where R: Read {
-            let reader = HttpReader::EofReader(data as &'a mut Read);
+        pub fn new<'a, 'b>(method: method::Method, path: Url,
+                           reader: &'a mut BufReader<&'b mut NetworkStream>) -> Request<'a, 'b> {
```
Comment (reem) [inline src/lib.rs]: I really dislike this API but I can't think of any other way to do this...

File: src/project_builder.rs [hunk 2]
```diff
@@ -31,7 +30,7 @@ impl FileBuilder {
                 .with_err_msg(format!("Could not create file; path={}",
                                       self.path.display())));
 
-        file.write_all(self.body.as_slice())
+        file.write_all(&*self.body)
```
Comment (reem) [inline src/project_builder.rs]: just `&` should work here I think

File: src/lib.rs [hunk 3]
```diff
@@ -18,6 +19,46 @@ mod project_builder;
...
+    }
+
+    impl<T> MockStream<T> {
+        /// Create a new mock stream that reads from the given data
+        pub fn new(data: T) -> MockStream<T> {
+            MockStream { data: data }
+        }
+    }
+
+    impl<T: Send + Read + Write + Clone + Reflect + 'static> NetworkStream for MockStream<T> {
```
Comment (reem) [inline src/lib.rs]: You can bound by `Any` instead of `Reflect + 'static` to get a little closer to only stable APIs.

**Comment by gsingh93** [comment]
I made the suggested changes and squashed the commits.

File: src/lib.rs (hunk 1 above)
Comment (gsingh93) [inline src/lib.rs]: Come to think of it, passing in a `MockStream` might have been better.

**Pull request merged**